### DISCUSSION TITLE
Bring back geom_brace

### DISCUSSION THREAD

**Discussion by wortmanncallejon** [NicolasH2/ggbrace#10] [General]
Hi! 

geom_brace was a really useful feature to label specific quantities as gaps in the x or y directions. I wish this feature was brought back. 

What is the reasoning for it being removed? Any chance you could bring it back?

**Comment by NicolasH2** [comment]
Hi @wortmanncallejon ,
thanks for the feedback. I understand that geom_brace was quite useful in its own right. The reason it is gone is that the same functionality can be achieved with stat_brace now. When you set `outside=FALSE` then stat_brace will have the same behavior as geom_brace before.

e.g.: `ggplot() + stat_brace(aes(x=c(1,2), y=c(1,2)), outside=FALSE)`

Does that resolve the issue? 
Again thanks for bringing this up. I will highlight it in the github readme :)

------small detour:
Apart from the all-in-one solution that stat_brace offers now, geom_brace was also discontinued because the name is misleading. In ggplot2, stat names are traditionally used when the function allows to change the geom part of the formula, while geom names are used when the stat can be changed.

**Reply by eahoegner** [reply]
Hi @NicolasH2, I just ran into problems because of the discontinuation as well. I was using geom_brace to add braces outside the plot for regions of the y-axis that i specified manually (independent from my data). Each brace was assigned a manual label, which in the previous version was very easily doable via geom_brace's aes(). I have now managed to rewrite my code to produce the braces, however, am not able to add the according labels with stat_bracetext, as stat_brace does not accept a label inside its aes(). Is there a solution I am overlooking? Thanks for any ideas!

**Reply by NicolasH2** [reply]
Hi @eahoegner,
for the labels, `stat_bracetext` should work just fine. You just have to supply the same parameters (e.g. `outside=FALSE`) to it. Can you share a minimal example of what does not work for you?

**Comment by eahoegner** [comment]
Hi @NicolasH2, sorry for my late reply. You are right, I was able to make it work in the end. Thanks again for the great package!

**Reply by Chris10102** [reply]
Hello， I am also trying to make a specific range of curly braces, but it always reports an error. Can you guide me?  
Here is my code and the error reported
stat_brace(aes(x = c(0.02, 0.99), y = c(71, 245.51)), 
             color = "blue", 
             size = 0.6,
             outside=FALSE) +
Error in `stat_brace()`:
! Problem while computing aesthetics.
ℹ Error occurred in the 3rd layer.
Caused by error in `check_aesthetics()`:
! Aesthetics must be either length 1 or the same as the data (408).
✖ Fix the following mappings: `x` and `y`.

**Reply by DanSimonet** [reply]
Same error as Chris101002. You need to enter the coordinates as a data frame and then put the values into the aes argument. I also vote for bringing back `geom_brace` as I was able to do something like the following with no issue which now requires extra steps

```
geom_brace(aes(c(.90, .95), c(19.90, 21.80)), inherit.data = F, rotate = 270) 
```

**Reply by NicolasH2** [reply]
Hey guys,
sorry for the super late reply. I have been absent for some time.
So the reason that geom_brace no longer exists is because in ggplot, geom is usually reserved for a visualization of the data points, whereas stat is a interpretation. As I understand it, it is a rather soft rule, but I nevertheless find it useful to distinguish what functions to choose. For the specific case of @DanSimonet, stat_brace actually doesn't need an extra step, just a bit more text:

`stat_brace(aes(x,y), data=data.frame(x=c(.90, .95), y=c(19.90, 21.80) ) )`
as opposed to your
`geom_brace(aes(c(.90, .95), c(19.90, 21.80) ) )`

since this is not even an extra line, I would not re-introduce a new function just for this purpse.

@Chris10102 , as DanSimonet mentioned, you need to wrap this into a data.frame, but it can be done within the stat_brace function:
```
stat_brace(aes(x,y), data=data.frame(x = c(0.02, 0.99), y = c(71, 245.51)),
           color = "blue",
           size = 0.6,
           outside=FALSE)
```
